src/mame/taito/taito_f3_v.cpp

```diff
@@ -2764,8 +2766,8 @@ void taito_f3_state::get_sprite_info(const u16 *spriteram16_ptr)
 						x = block_x;
 					else
 					{
-						this_x = spriteram16_ptr[current_offs + 2 + 0];
-						if (this_x & 0x800) this_x = 0 - (0x800 - (this_x & 0x7ff)); else this_x &= 0x7ff;
+						this_x = spriteram16_ptr[current_offs + 2 + 0] & 0xfff;
+						if (this_x & 0x800) this_x -= 0x1000; // 12->16 sign extension
```

Comment:
You can use `util::sext` for sign extension.

Reply:
This has been changed to
```
this_x = spriteram16_ptr[current_offs + 2 + 0] & 0xfff;
this_x = util::sext(this_x, 12);
```
Question: should the `& 0xfff` stay still?

Reply:
I don’t think it’s necessary – `util::sext` is supposed to completely replace what’s in the upper bits.  Leaving it there isn’t going to hurt though.